### DISCUSSION TITLE
Possible to detect single "component:remove" event despite having child components?

### DISCUSSION THREAD

**Discussion by protozoo** [GrapesJS/grapesjs#4736] [Q&A] [answered]
Hi there,

I am trying to detect when the user removes a component from the canvas by listening to the "component:remove" event. This works fine for simple components but will trigger multiple times if the removed component has child components. Is there a way to capture just a single event? (I am interested in capturing the user action of removing something, regardless of the chain on removals that this action will trigger).

Thanks!

**Comment by artf** [comment, marked as answer]
Would it work for your case checking the selected component?
```js
editor.on('component:remove', (removed) => {
 if (removed === editor.getSelected()) {
   // ...
 }
})
```

**Reply by protozoo** [reply]
Yes, this solves the the issue, many thanks
